tokio/src/task/builder.rs

```diff
@@ -72,7 +75,18 @@ impl<'a> Builder<'a> {
 
     /// Assigns a name to the task which will be spawned.
     pub fn name(&self, name: &'a str) -> Self {
-        Self { name: Some(name) }
+        Self {
+            name: Some(name),
+            user_data: self.user_data,
+        }
+    }
+
+    /// Assigns user data to the task which will be spawned.
+    pub fn data(self, data: &'static dyn std::any::Any) -> Self {
```

Comment:
nit: this method consumes `self`, while `fn name(&self)` above does not. I think `name()` should be changed too.

Reply:
So you're right in your other comment that the builder is not behnd unstable, which means changing `name()` is a breaking API change on a stable API. I'll change `data()` to match `name()` instead.